.circleci/config.yml

```diff
@@ -264,136 +402,98 @@ jobs:
-          command: |
-            ../ci/activate_creds.sh circle-sa-key.json
-            ./project.rb deploy \
-              --project all-of-us-rw-staging \
-              --account circle-deploy-account@all-of-us-workbench-test.iam.gserviceaccount.com \
-              --git-version "${CIRCLE_TAG}" \
-              --app-version "${CIRCLE_TAG}" \
-              --circle-url "https://circleci.com/workflow-run/${CIRCLE_WORKFLOW_ID}" \
-              --key-file circle-sa-key.json \
-              --promote
+      - git_checkout
+      - restore_api_cache
+      - activate_service_accont_creds
+      - deploy_appengine:
+          project_name: "all-of-us-rw-staging"
+          deploy_arg: "deploy"
+          working_dir: ~/workbench/deploy
 
   deploy-perf:
-    <<: *defaults
+    executor: workbench_docker
     steps:
-      - checkout
+      - git_checkout
+      - ensure_branch_has_api_changes
+      - restore_api_cache
+      - activate_service_accont_creds
+      - deploy_appengine:
+          project_name: "all-of-us-rw-perf"
+          deploy_arg: "deploy"
+          working_dir: ~/workbench/deploy
+
+  # deploy UI and API local server
+  deploy-local:
+    executor: workbench_docker
+    steps:
+      - start_api_local:
+          skip_no_api_change: false
+      - attach_workspace:
+          at: .
+      # - git_checkout
+      - ui_dependencies
+      - restore_api_cache
       - run:
-          command: git submodule update --init --recursive
-      - restore_cache:
-          keys:
-          - api-cache-{{ checksum "~/workbench/api/build.gradle" }}
-          - api-cache-
-      - deploy:
-          name: Deploy to Perf Environment
-          working_directory: ~/workbench/deploy
+          name: Launch local UI server
+          working_directory: ~/workbench/ui
+          # Start local UI server and running in the background
+          command: yarn dev-up --configuration=local
+          background: true
+      - run:
+          name: Wait for localhost UI to be started
           command: |
-            ../ci/activate_creds.sh circle-sa-key.json
-            ./project.rb deploy \
-              --project all-of-us-rw-perf \
-              --account circle-deploy-account@all-of-us-workbench-test.iam.gserviceaccount.com \
-              --git-version "${CIRCLE_TAG}" \
-              --app-version "${CIRCLE_TAG}" \
-              --circle-url "https://circleci.com/workflow-run/${CIRCLE_WORKFLOW_ID}" \
-              --key-file circle-sa-key.json \
-              --promote
+            sleep 2m
+            dockerize -wait tcp://127.0.0.1:4200 -timeout 3m
+
+
+  # Run Puppeteer end-to-end UI tests on local API & UI server.
+  e2e-test-local:
+    executor: browser_docker
+    environment:
+      WORKBENCH_ENV: local
+    parallelism: 2
+    steps:
+      - attach_workspace:
+          at: .
+      - e2e_tests
+
+  # Run Puppeteer end-to-end UI tests on Workbench "test" environment.
+  e2e-test:
```

Comment:
The name should be `e2e-test-test` I think. Or make `e2e-test-local` just be `e2e-local`. It's unfortunate our qa environment is called test.